code/game/objects/items/crab17.dm

```diff
@@ -58,28 +58,35 @@
 			return FALSE
 	return TRUE
 
-/obj/structure/checkoutmachine/attackby(obj/item/W, mob/user, params)
+/obj/structure/checkoutmachine/attackby(obj/item/attacking_item, mob/user, params)
+	if(check_if_finished())
+		qdel(src)
+		return
+
+	var/obj/item/card/id/card = attacking_item.GetID()
+	if(!card)
+		balloon_alert(user, "your [attacking_item.name] gets repelled by the id card reader")
```

Comment:
For posterity--usually balloon alerts saying the item name is inappropriate, but I'm fine with it for this.

Reply:
```suggestion
		balloon_alert(user, "your [attacking_item.name] gets repelled by the id card reader!")
```

My only adjustment here--`!` indicates failure